xarray/core/dataset.py

```diff
@@ -2416,6 +2416,60 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=False):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Both the data and coordinate variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, set_index only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+
+        ordered_dims = self.dims
+        coord_columns = [c for c in self if c in ordered_dims]
+
+        # convert all coordinates into data variables
+        ds = self.reset_index(coord_columns).reset_coords()
+
+        # ensure we are dealing with dask arrays with consistent chunks
+        chunked = ds.chunk(chunks=self.chunks)
+
+        # order columns so that coordinates appear before data
+        columns = list(chunked)
+        num_coords = len(coord_columns)
+        columns = columns[-num_coords:] + columns[:-num_coords]
+
+        data = [chunked._variables[k].set_dims(ordered_dims).data.reshape(-1)
+                for k in columns]
+
+        # restore names of columns that were also index names (e.g. x_ -> x)
+        columns = [c[:-1] if c[:-1] in ds.dims else c for c in columns]
+
+        df = dd.concat([dd.from_array(d, columns=c)
+                        for d, c in zip(data, columns)], axis=1)
+
+        if set_index:
+
+            if len(ordered_dims) != 1:
+                raise ValueError(
+                        'set_index=True only is valid for '
+                        'for one-dimensional datasets')
+
+            # extract out first (and only) coordinate variable
+            coord_dim = list(ordered_dims)[0]
+
+            df = df.set_index(coord_dim)
+
+        if isinstance(df, dd.Series):
+            df = df.to_frame()
```

Comment:
Does `dask.dataframe.concat` really create a Series if asked to concat along `axis=1` with only one input? If so, it would be appreciated if you could file this as an dask bug (which it almost certainly is) and reference it here.

Reply:
Yes, that appears to be the current behaviour of dask.dataframe.concat.  And I agree that this not the same as what pd.concat does (if s is a pandas Series, then pd.concat([s], axis=1) is indeed a DataFrame).

Reply:
Issue filed as  dask/dask#2798